### DISCUSSION TITLE
Formula to exclude certain clusters?

### DISCUSSION THREAD

**Discussion by yabeykoon** [Katsevich-Lab/sceptre#129] [Q&A] [answered]
Hi, Thank you for SCEPTRE. It has helped me immensely to analyze my high MOI perturb-seq data! From my single cell run, I have some cells that seemed to have bulked during the capture, that form a few low quality clusters in my UMAP when I analyze my cellranger output using Scanpy. Is there a way to eliminate certain low quality clusters by adding them as covariates in the SCEPTRE pipeline? I would appreciate an example code if so.

**Comment by ekatsevi** [comment, marked as answer]
> Thank you for SCEPTRE. It has helped me immensely to analyze my high MOI perturb-seq data! 

That's great to hear! 

> Is there a way to eliminate certain low quality clusters by adding them as covariates in the SCEPTRE pipeline?

The best way to do this is to extract a (one-based) integer vector of indices of cells in low-quality clusters from Scanpy, and then to pass this to the `additional_cells_to_remove` argument of the [`run_qc()`](https://katsevich-lab.github.io/sceptre/reference/run_qc.html) function.

**Reply by yabeykoon** [reply]
Thank you! Is it possible to add this step to the Nextflow trans analysis? I don't see where I could add this argument in the Nextflow pipeline.
